skops/cli/_convert.py

```diff
@@ -0,0 +1,114 @@
+from __future__ import annotations
+
+import argparse
+import logging
+import os
+import pathlib
+import pickle as pkl
+from typing import Optional
+
+from skops.io import dumps, get_untrusted_types
+
+
+def _convert_file(input_file: os.PathLike, output_file: pathlib.Path):
+    """
+    Function that is called by ``skops convert`` entrypoint.
+
+    Loads a pickle model from the input path, converts to skops format, and saves to
+    output file.
+
+    Parameters
+    ----------
+    input_file : os.PathLike
+        Path of input .pkl model to load.
+
+    output_file : os.PathLike
+        Path to save .skops model to.
+
+    """
+    model_name = pathlib.Path(input_file).stem
+
+    logging.info(f"Converting {model_name}")
+
+    with open(input_file, "rb") as f:
+        obj = pkl.load(f)
+    skops_dump = dumps(obj)
+
+    untrusted_types = get_untrusted_types(data=skops_dump)
+
+    if not untrusted_types:
+        logging.debug(f"No unsafe types found in {model_name}.")
+    else:
+        untrusted_str = "\n".join(untrusted_types)
+
+        logging.warning(
+            "Unknown Types Detected!\n"
+            f"While converting {model_name}, "
+            "the following unsafe types were found: \n"
+            f"{untrusted_str}\n\n"
+            f"When loading{output_file}, add ``--trusted`` to the skops.load call."
+        )
+
+    with open(output_file, "wb") as out_file:
+        logging.info(f"Writing to {output_file}")
+        out_file.write(skops_dump)
+
+
+def main_convert(
+    command_line_args: Optional[list[str]] = None,
+    parent: Optional[argparse.ArgumentParser] = None,
+):
+    parents = [parent] if parent else []
+
+    parser = argparse.ArgumentParser(
+        description="Convert input Pickle files to .skops files", parents=parents
+    )
+
+    parser.add_argument("inputs", nargs="+", help="Input files to convert.")
```

Comment:
I think starting with a single file would make more sense and makes things much easier. We also don't have to deal with matching number of inputs/outputs etc. We can let a shell script handle multiple files.

Reply:
I think that would be ok, though I think there should still be support for converting batches of files at a time. 

I can rework this PR to work for single files only, but I do feel we should consider ways to support it.

Reply:
Let's start with single files. We can maybe document how to convert multiple files to make it easier for users to achieve that. Then, when we run into a case where batches would be really useful, we can add them later.